### PR TITLE
MKC: prisoners_dilemma + support

### PR DESCRIPTION
**Pull request by Northmoc**
AI currently only chooses Silence... probably "good enough for now" ?

### CODE REVIEW

**Comment by Agetian** [comment]
Hmm yeah, not sure what logic could work here, probably good enough for now :)

**Review by Agetian — DISMISSED** [review]

File: forge-game/src/main/java/forge/util/MessageUtil.java [hunk 1]
```diff
@@ -63,7 +63,8 @@ public static  String formatNotificationMessage(SpellAbility sa, Player player,
                         ? Localizer.getInstance().getMessage("lblPlayerFlipComesUpValue", Lang.getInstance().getPossesive(flipper), value)
                         : Localizer.getInstance().getMessage("lblPlayerActionFlip", flipper, Lang.joinVerb(flipper, value));
             case GenericChoice:
-                if (sa.hasParam("ShowChoice") && sa.getParam("ShowChoice").equals("Description")) {
+                if (sa.hasParam("Secretly") ||
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/util/MessageUtil.java:66]: better group the parts with parenthesis to guard against logic flaws

File: forge-game/src/main/java/forge/game/ability/effects/ChooseGenericEffect.java [hunk 2]
```diff
@@ -49,6 +50,8 @@ public void resolve(SpellAbility sa) {
         final int amount = AbilityUtils.calculateAmount(host, sa.getParamOrDefault("ChoiceAmount", "1"), sa);
         
         final boolean tempRem = sa.hasParam("TempRemember");
+        final boolean secretly = sa.hasParam("Secretly");
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/ability/effects/ChooseGenericEffect.java:53]: should be "Secret" for consistency with other

Comment (Northmoc) [reply forge-game/src/main/java/forge/game/ability/effects/ChooseGenericEffect.java:53]: ChooseTypeEffect and ChoosePlayerEffect also had "Secretly" so I just changed VoteEffect

**Review by tool4ever — APPROVED** [review]

**Pull request merged**